plugin/src/android/src/main/java/com/mparticle/cordova/MParticleCordovaPlugin.java

```diff
@@ -326,6 +338,56 @@ public void getUserIdentities(final JSONArray args, final CallbackContext callba
         callbackContext.sendPluginResult(pluginResult);
     }
 
+    public void addGDPRConsentState(final JSONArray args) throws JSONException {
+        MParticleUser user = MParticle.getInstance().Identity().getCurrentUser();
+        final JSONObject map = args.getJSONObject(0);
```

Comment:
So, I'm not going to block the PR because it looks like we already do this in a bunch of other methods, but the `JSONArray.getJSONObject(int)` method (or any JSONObject/JSONArray `get{type}()` method) will throw an exception if the value is not a `JSONObject` (or `{type}`), meaning our null-check on this value won't every help us.

The most defensive way to handle this parsing would be like:

```
JSONObject map = null;
if (args.length > 0) {
    map = args.optJSONObject(0)
}
String purpose = null;
if (args.length > 1) {
    purpose = args.optString(1)
}
if (user != null && map != null && purpose != null) {
    //do the stuff
}
```

Since your implementation is consistent with how we do it elsewhere in this file, it'd be fine to get this out and fast-follow with a broader update